src/EpisodeIntegrationService/ReceiveData/ReceiveData.cs

```diff
@@ -162,6 +178,35 @@ private EpisodeDto MapEpisodeToEpisodeDto(BssEpisode episode)
         };
     }
 
+    private DateOnly? ParseNullableDate(string? date)
+    {
+        if (string.IsNullOrEmpty(date)) return null;
+
+        if (DateTime.TryParseExact(date, AllowedDateFormats, CultureInfo.InvariantCulture, DateTimeStyles.None, out var dateTime))
+        {
+            return DateOnly.FromDateTime(dateTime);
+        }
+
+        throw new FormatException($"Invalid date format: {date}. Expected formats: {string.Join(", ", AllowedDateFormats)}");
```

Comment:
If we use ParseExact rather than TryParseExact then we would be able to avoid throwing a FormatException as ParseExact will do it for us